reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java

```diff
@@ -814,6 +814,10 @@ else if (msg instanceof HttpRequest) {
 				}
 				else {
 					request.release();
+					// HTTP/1.1 TLS Upgrade (RFC-2817) on empty requests (GET/HEAD/OPTIONS)
```

Comment:
It is true that GET should be used without payload, however there are use cases when GET is actually used with payload. (I do not refer to this HTTP/1.1 TLS Upgrade, but in general)

What will happen if we have this request:

```
		createHttpClientForContextWithPort()
		        .protocol(HttpProtocol.HTTP11)
		        .headers(h ->
		            h.add(HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE)
		             .add(HttpHeaderNames.UPGRADE, "TLS/1.2"))
		        .request(HttpMethod.GET)
		        .send((req, res) -> res.sendString(Mono.just("testIssue3538")))
                ...
```

Reply:
👍 , I will add test case for it, I would expect `onInboundNext()` call for sure, but `onInboundComplete()` is under the question

Reply:
Turned out to be more nuanced than that: 
 - by default, the request is rejected with 413
 - the default could be overridden with `.httpRequestDecoder(spec -> spec.h2cMaxContentLength(...))`

I have a test case for the latter but it fails with 

```
reactor.core.Exceptions$ReactiveException: reactor.netty.http.client.PrematureCloseException: Connection prematurely closed DURING response
```

Trying to figure out why.

Reply:
Resolved, @violetagg we should be all set, thank you!